src/Mvc/Mvc.Core/src/Routing/ControllerEndpointDataSourceIdMetadata.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    internal class ControllerEndpointDataSourceIdMetadata
+
```

Comment:
It's explained below, but essentially I don't want to put the DataSource within the endpoint to prevent people from discovering it if they inspect the metadata and doing stuff with it.